exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillDatabaseMetaDataImpl.java

```diff
@@ -184,6 +184,17 @@ private ServerMeta getServerMeta() throws SQLException {
     return serverMeta;
   }
 
+  /**
+   * The same as {@link DrillDatabaseMetaDataImpl#getServerMeta()} but updates server metadata for the every call
+   * @return server meta information
+   * @throws SQLException for error when getting server meta
+   */
+  private ServerMeta getUpdatedServerMeta() throws SQLException {
```

Comment:
SQL Server, for example, updates metadata in some cases
https://www.mssqltips.com/sqlservertip/1415/determining-set-options-for-a-current-session-in-sql-server/
Quoting_identifiers can be changes as system option. In this case any user can get the right and actual quoting_identifiers method from server_metadata.